Flaky test - github.com/docker/buildx/dap

### Contributing guidelines

- [x] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

### Description

When building buildx v0.32.0 on Fedora Rawhide (go 1.26) I am seeing a flaky test result. In the `go test` section of the output on both x86_64 and ppc64le these result appears:

```
github.com/docker/buildx/dap
panic: sync: WaitGroup is reused before previous Wait has returned
goroutine 28 [running]:
sync.(*WaitGroup).Wait(0x45b5d41e2208)
	/usr/lib/golang/src/sync/waitgroup.go:213 +0x198
github.com/docker/buildx/vendor/golang.org/x/sync/errgroup.(*Group).Wait(0x45b5d41e2200)
	/builddir/build/BUILD/docker-buildx-0.32.0-build/buildx-0.32.0/_build/src/github.com/docker/buildx/vendor/golang.org/x/sync/errgroup/errgroup.go:56 +0x4c
github.com/docker/buildx/dap.(*Server).Serve.func4()
	/builddir/build/BUILD/docker-buildx-0.32.0-build/buildx-0.32.0/_build/src/github.com/docker/buildx/dap/server.go:61 +0x78
github.com/docker/buildx/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1()
	/builddir/build/BUILD/docker-buildx-0.32.0-build/buildx-0.32.0/_build/src/github.com/docker/buildx/vendor/golang.org/x/sync/errgroup/errgroup.go:93 +0x74
created by github.com/docker/buildx/vendor/golang.org/x/sync/errgroup.(*Group).Go in goroutine 24
	/builddir/build/BUILD/docker-buildx-0.32.0-build/buildx-0.32.0/_build/src/github.com/docker/buildx/vendor/golang.org/x/sync/errgroup/errgroup.go:78 +0xcc
exit status 2
FAIL	github.com/docker/buildx/dap	0.036s
```

Complete test results available at: https://koji.fedoraproject.org/koji/taskinfo?taskID=143024149

I have triggered this several times with the same result. However, I have no problems doing a build in a chroot of rawhide on my workstation. Other architectures do not fail.

Potentially related to https://github.com/docker/buildx/issues/3598

### Expected behaviour

Successful build

### Actual behaviour

Failure in `go test` as noted. Failure flaky in that it does not occur on all architectures and will succeed on x86_64 workstation but fail on x86_64 builder.

### Buildx version

0.32.0 

### Docker info

```text
n/a error during build
```

### Builders list

```text
n/a error during build
```

### Configuration

n/a



### Build logs

```text
https://koji.fedoraproject.org/koji/taskinfo?taskID=143024149

See, for example, https://kojipkgs.fedoraproject.org//work/tasks/4202/143024202/build.log
```

### Additional info

n/a.

--- a/commands/dap.go
+++ b/commands/dap.go
@@ -1,7 +1,6 @@
 package commands
 
 import (
-	"context"
 	"io"
 	"net"
 	"os"
@@ -75,7 +74,7 @@ type adapterProtocolDebugger struct {
 }
 
 func (d *adapterProtocolDebugger) Start(printer *progress.Printer, opts *BuildOptions) error {
-	cfg, err := d.Adapter.Start(context.Background(), d.conn)
+	cfg, err := d.Adapter.Start(d.conn)
 	if err != nil {
 		return errors.Wrap(err, "debug adapter did not start")
 	}

--- a/dap/adapter.go
+++ b/dap/adapter.go
@@ -67,10 +67,10 @@ func New[C LaunchConfig]() *Adapter[C] {
 	return d
 }
 
-func (d *Adapter[C]) Start(ctx context.Context, conn Conn) (C, error) {
-	d.eg, _ = errgroup.WithContext(ctx)
+func (d *Adapter[C]) Start(conn Conn) (C, error) {
+	d.eg, _ = errgroup.WithContext(context.Background())
 	d.eg.Go(func() error {
-		return d.srv.Serve(ctx, conn)
+		return d.srv.Serve(conn)
 	})
 
 	<-d.initialized

--- a/dap/adapter_test.go
+++ b/dap/adapter_test.go
@@ -24,7 +24,7 @@ func TestLaunch(t *testing.T) {
 
 	eg, _ := errgroup.WithContext(ctx)
 	eg.Go(func() error {
-		_, err := adapter.Start(ctx, conn)
+		_, err := adapter.Start(conn)
 		assert.NoError(t, err)
 		return nil
 	})
@@ -57,13 +57,13 @@ func TestLaunch(t *testing.T) {
 		// We should have received the initialized event.
 		select {
 		case <-initialized:
-		default:
+		case <-ctx.Done():
 			assert.Fail(t, "did not receive initialized event")
 		}
 
 		select {
 		case <-configurationDone:
-		case <-time.After(10 * time.Second):
+		case <-ctx.Done():
 			assert.Fail(t, "did not receive configurationDone response")
 		}
 		return nil
@@ -82,7 +82,7 @@ func TestSetBreakpoints(t *testing.T) {
 
 	eg, _ := errgroup.WithContext(ctx)
 	eg.Go(func() error {
-		_, err := adapter.Start(ctx, conn)
+		_, err := adapter.Start(conn)
 		assert.NoError(t, err)
 		return nil
 	})
@@ -118,7 +118,7 @@ func TestSetBreakpoints(t *testing.T) {
 		// We should have received the initialized event.
 		select {
 		case <-initialized:
-		default:
+		case <-ctx.Done():
 			assert.Fail(t, "did not receive initialized event")
 		}
 
@@ -127,7 +127,7 @@ func TestSetBreakpoints(t *testing.T) {
 			assert.True(t, setBreakpointsResp.Success)
 			assert.Len(t, setBreakpointsResp.Body.Breakpoints, 0)
 			assert.NotNil(t, setBreakpointsResp.Body.Breakpoints, "breakpoints should be an empty array instead of null in the JSON")
-		case <-time.After(10 * time.Second):
+		case <-ctx.Done():
 			assert.Fail(t, "did not receive setBreakpoints response")
 		}
 		return nil
@@ -249,11 +249,11 @@ func NewTestAdapter[C LaunchConfig](t *testing.T) (*Adapter[C], Conn, *daptest.C
 	t.Cleanup(func() { clientConn.Close() })
 
 	adapter := New[C]()
-	t.Cleanup(func() { adapter.Stop() })
 
 	client := daptest.NewClient(clientConn)
 	t.Cleanup(func() { client.Close() })
 
+	t.Cleanup(func() { adapter.Stop() })
 	return adapter, srvConn, client
 }
 

--- a/dap/server.go
+++ b/dap/server.go
@@ -33,11 +33,11 @@ func NewServer(h Handler) *Server {
 	return &Server{h: h}
 }
 
-func (s *Server) Serve(ctx context.Context, conn Conn) error {
+func (s *Server) Serve(conn Conn) error {
 	writeCh := make(chan dap.Message)
 	s.ch = writeCh
 
-	s.ctx, s.cancel = context.WithCancelCause(ctx)
+	s.ctx, s.cancel = context.WithCancelCause(context.Background())
 
 	// Start an error group to handle server-initiated tasks.
 	s.eg, _ = errgroup.WithContext(s.ctx)

